plugins/zoom/icon_zoom/unit_test/test_monitor_sign_in_out_activity.py

```diff
@@ -0,0 +1,411 @@
+import logging
+import unittest
+import datetime
+from unittest.mock import patch
+
+from icon_zoom.util.event import Event
+from icon_zoom.tasks.monitor_sign_in_out_activity.task import MonitorSignInOutActivity
+from icon_zoom.connection.connection import Connection
+from icon_zoom.unit_test.mock import STUB_CONNECTION
+
+GET_USER_ACTIVITY_EVENTS_PATH = "icon_zoom.util.api.ZoomAPI.get_user_activity_events"
+GET_DATETIME_NOW_PATH = "icon_zoom.tasks.monitor_sign_in_out_activity.task.MonitorSignInOutActivity._get_datetime_now"
+
+
+class TestGetUserActivityEvents(unittest.TestCase):
+    def setUp(self) -> None:
+        self.connection = Connection()
+        self.connection.logger = logging.getLogger("connection logger")
+        self.connection.connect(STUB_CONNECTION)
+
+        self.action = MonitorSignInOutActivity()
+        self.action.connection = self.connection
+        self.action.logger = logging.getLogger("action logger")
+        self.params = {}
+
+    @patch(GET_USER_ACTIVITY_EVENTS_PATH)
+    def test_first_run(self, mock_call):
+        expected_output = [
+            {
+                "client_type": "mac",
```

Comment:
(Suggestion): Move the expected output values into a separate file and just pull them in so the test files are less cluttered/

Reply:
We're going to be rewriting a good portion of the task once pagination is supported by our external services - can this wait until that rewrite happens? I'll have to update all of the tests for the rewrite.

Reply:
Ok thats fine.